drivers/android-driver/src/main/java/app/cash/sqldelight/driver/android/AndroidSqliteDriver.kt

```diff
@@ -200,7 +201,23 @@ class AndroidSqliteDriver private constructor(
         oldVersion,
         newVersion,
         *callbacks,
-      )
+      ).requireSynchronous(schema)
+    }
+
+    private fun QueryResult<*>.requireSynchronous(schema: SqlSchema) {
+      if (this is QueryResult.AsyncValue) {
+        throw IllegalStateException(
+          """
+          |The android driver is synchronous, but you configured SQLDelight to be asynchronous. This
+          |will result in unexpected behavior since suspending functions would actually block. If
+          |the generated code must be asynchronous (ie, because it is being used by another driver
+          |which must be asynchronous), you can get around this error by passing a synchronous schema
+          |to this driver:
+          |
+          |AndroidSqliteDriver(${schema::class.simpleName}.synchronous(), context, ...)
+          """.trimMargin(),
+        )
+      }
```

Comment:
```suggestion
      check (this !is QueryResult.AsyncValue) {
        """
        |The android driver is synchronous, but you configured SQLDelight to be asynchronous. This
        |will result in unexpected behavior since suspending functions would actually block. If
        |the generated code must be asynchronous (ie, because it is being used by another driver
        |which must be asynchronous), you can get around this error by passing a synchronous schema
        |to this driver:
        |
        |AndroidSqliteDriver(${schema::class.simpleName}.synchronous(), context, ...)
        """.trimMargin()
      }
```

`check() {}` over `if() { throw }`?